nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/PublishMQTT.java

```diff
@@ -0,0 +1,288 @@
+
+    private Set<Relationship> relationships;
+    private ProcessorLog logger;
+    private MqttConnectOptions connOpts;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        logger = getLogger();
+        final List<PropertyDescriptor> descriptors = getAbstractPropertyDescriptors();
+        descriptors.add(PROP_TOPIC);
+        descriptors.add(PROP_QOS);
+        descriptors.add(PROP_RETAIN);
+        descriptors.add(PROP_TIMEOUT);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        try {
+            broker = context.getProperty(PROP_BROKER_URI).getValue();
+            String clientID = context.getProperty(PROP_CLIENTID).getValue();
+
+            connOpts = new MqttConnectOptions();
+            connOpts.setCleanSession(true);
+
+            PropertyValue sslProp = context.getProperty(PROP_SSL_CONTEXT_SERVICE);
+            if (sslProp.isSet()) {
+                Properties sslProps = transformSSLContextService((SSLContextService) sslProp.asControllerService());
+                connOpts.setSSLProperties(sslProps);
+            }
+
+            PropertyValue lastWillTopicProp = context.getProperty(PROP_LAST_WILL_TOPIC);
+            if (lastWillTopicProp.isSet()){
+                PropertyValue lastWillMessage = context.getProperty(PROP_LAST_WILL_TOPIC);
+                PropertyValue lastWillRetain = context.getProperty(PROP_LAST_WILL_TOPIC);
+                connOpts.setWill(lastWillTopicProp.getValue(), lastWillMessage.getValue().getBytes(), LAST_WILL_QOS, lastWillRetain.isSet() ? lastWillRetain.asBoolean() : false);
+            }
+
+            PropertyValue usernameProp = context.getProperty(PROP_USERNAME);
+            if(usernameProp.isSet()) {
+                connOpts.setUserName(usernameProp.getValue());
+                connOpts.setPassword(context.getProperty(PROP_PASSWORD).getValue().toCharArray());
+            }
+
+            synchronized (mqttClientLock) {
+                mqttClient = new MqttClient(broker, clientID, persistence);
+
+                mqttClient.setCallback(new PublishMQTTCallback());
+                getLogger().info("Connecting to broker: " + broker);
+                mqttClient.connect(connOpts);
+            }
+        } catch(MqttException me) {
+            logger.error("Failed to initialize the connection to the  "+me.getMessage());
+        }
+    }
```

Comment:
Also, although I am still reviewing I am assuming you have to obtain connection in both processors, which means some of this logic may be better in the abstract base class